From a625a8689e7332cdcb2ba418731fd641bdb13c00 Mon Sep 17 00:00:00 2001
From: The-going <48602507+The-going@users.noreply.github.com>
Date: Fri, 27 May 2022 14:45:59 +0300
Subject: [PATCH 156/170] fix: cpu opp table sun8i-a83t

cpu opp table can only be a multiple of the integer 24 * 8 * n
---
 arch/arm/boot/dts/sun8i-a83t.dtsi | 52 +++++++++++++++++++------------
 1 file changed, 32 insertions(+), 20 deletions(-)

diff --git a/arch/arm/boot/dts/sun8i-a83t.dtsi b/arch/arm/boot/dts/sun8i-a83t.dtsi
index f0711d68b..4fad00e0e 100644
--- a/arch/arm/boot/dts/sun8i-a83t.dtsi
+++ b/arch/arm/boot/dts/sun8i-a83t.dtsi
@@ -224,32 +224,38 @@ cpu0_opp_table: opp-table-cluster0 {
 		compatible = "operating-points-v2";
 		opp-shared;
 
-		opp-480000000 {
-			opp-hz = /bits/ 64 <480000000>;
+		opp-576000000 {
+			opp-hz = /bits/ 64 <576000000>; /* 24x8x3 */
 			opp-microvolt = <840000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
-		opp-1008000000 {
-			opp-hz = /bits/ 64 <1008000000>;
+		opp-768000000 {
+			opp-hz = /bits/ 64 <768000000>; /* 24x8x4 */
 			opp-microvolt = <840000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
-		opp-1412000000 {
-			opp-hz = /bits/ 64 <1412000000>;
+		opp-1152000000 {
+			opp-hz = /bits/ 64 <1152000000>; /* 24x8x6 */
+			opp-microvolt = <900000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp-1344000000 {
+			opp-hz = /bits/ 64 <1344000000>; /* 24x8x7 */
 			opp-microvolt = <920000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
-		opp-1608000000 {
-			opp-hz = /bits/ 64 <1608000000>;
+		opp-1536000000 {
+			opp-hz = /bits/ 64 <1536000000>; /* 24x8x8 */
 			opp-microvolt = <1000000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
-		opp-1800000000 {
-			opp-hz = /bits/ 64 <1800000000>;
+		opp-1728000000 {
+			opp-hz = /bits/ 64 <1728000000>; /* 24x8x9 */
 			opp-microvolt = <1080000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
@@ -259,32 +265,38 @@ cpu1_opp_table: opp-table-cluster1 {
 		compatible = "operating-points-v2";
 		opp-shared;
 
-		opp-480000000 {
-			opp-hz = /bits/ 64 <480000000>;
+		opp-576000000 {
+			opp-hz = /bits/ 64 <576000000>; /* 24x8x3 */
 			opp-microvolt = <840000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
-		opp-1008000000 {
-			opp-hz = /bits/ 64 <1008000000>;
+		opp-768000000 {
+			opp-hz = /bits/ 64 <768000000>; /* 24x8x4 */
 			opp-microvolt = <840000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
-		opp-1412000000 {
-			opp-hz = /bits/ 64 <1412000000>;
+		opp-1152000000 {
+			opp-hz = /bits/ 64 <1152000000>; /* 24x8x6 */
+			opp-microvolt = <900000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp-1344000000 {
+			opp-hz = /bits/ 64 <1344000000>; /* 24x8x7 */
 			opp-microvolt = <920000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
-		opp-1608000000 {
-			opp-hz = /bits/ 64 <1608000000>;
+		opp-1536000000 {
+			opp-hz = /bits/ 64 <1536000000>; /* 24x8x8 */
 			opp-microvolt = <1000000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
-		opp-1800000000 {
-			opp-hz = /bits/ 64 <1800000000>;
+		opp-1728000000 {
+			opp-hz = /bits/ 64 <1728000000>; /* 24x8x9 */
 			opp-microvolt = <1080000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
-- 
2.35.3

